content/en/cd-as-a-service/concepts/deployment/how-hpa-works.md

```diff
@@ -0,0 +1,73 @@
+
+ An HPA resource has three components:
+- A set of one or more metrics, each with a desired target value.
+- A reference to a Kubernetes resource, typically a Deployment but any resource that supports the Scale API sub-resource (e.g., `ReplicaSet`, `StatefulSet`). The HPA scales this resource in response to changes in observed metrics.
+- A minimum and maximum number of replicas.
+
+HPA checks metrics as configured in the deployment file AT A DEFAULT 30 SEC intervals. When the threshold load is met the HPA increases or decreases the number of pods. 
+ 
+ Kubernetes implements Horizontal Auto Scaling (HPA) as a control loop that runs intermittently. `HorizontalPodAutoscaler` controls the scale of a Deployment and its `ReplicaSet`. 
+ 
+ HPA is not recommended for replication controllers. 
+ 
+ HPA does not manage rolling updates by manipulating the replication controller. The deployment object manages the size of underlying replica sets. 
+
+ The sync interval is set by the `kube-controller-manager`. The default interval is 30 seconds. Change the default interval using the `horizontal-pod-autoscaler-sync-period` flag.
+
+The default delay is 3 minutes after an upscale event to allow metrics to stabilize. Change the default upscale delay period using the `horizontal-pod-autoscaler-upscale-delay` flag.
+ 
+## How HPA works in CD-as-a-Service
+When HPA is configured for an Armory Continuous Deployment-as-a-Service (CD-as-a-Service) deployment the deployment is converted to a `ReplicaSet` and the HPA configuration file is automatically re-written to reference the generated `ReplicaSet` resource. Continuous Deployment-as-a-Service freezes scaling behavior during a deployment.
+
+Consider a deployment to upgrade from the v1 to v2 version of your application. In this scenario the v1 application is running with 10 replicas and is scaled by HPA (min: 5 pods, max: 15 pods).
+ - At the beginning of a deployment, the HPA is deleted. Your v1 application has 10 pods but no longer scales between 5 and 15 pods.
+ - CD as a Service deploys the v2 instance of your application with 10 pods (20 total pods between v1 and v2).
+ - At the end of the deployment, a new HPA resource is created to target the v2 application.
+ - If the deployment needs to be rolled back, CDaaS deletes the v2 application and re-creates the original HPA.
+
+### Supported versions
+Armory Continuous Deployment-as-a-Service supports the following HorizontalPodAutoscaler API versions:
+- [v1](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v1/)
+- [v2beta 1 and 2](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2beta2/)
+-  [v2](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2/)
```

Comment:
Great that you added the links. What do you think about moving Supported Versions section to a "##" near or at the bottom?

Reply:
I did this in my re-write - totally agree